compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -907,7 +908,7 @@ private void checkFunctionInvocationExpr(BLangInvocation iExpr) {
             // Check for function pointer.
             iExpr.functionPointerInvocation = true;
         }
-
+        checkFunctionDeprecated(iExpr.pos, names.fromIdNode(iExpr.name), funcSymbol);
```

Comment:
We shouldn't do this type of warning logging in the type checker. This has to go the code analyzer phase. 
But please create an issue to track this for now. and put the issue ID in the code comment.